File: HaberUygulamaJava/app/src/main/java/com/example/haberuygulamajava/view/KategoriFragment.java
package com.example.haberuygulamajava.view;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleEventObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.haberuygulamajava.R;
import com.example.haberuygulamajava.adapter.HaberlerRVAdapter;
import com.example.haberuygulamajava.dao.HaberDao;
import com.example.haberuygulamajava.databinding.FragmentKategoriBinding;
import com.example.haberuygulamajava.model.HaberModel;
import com.example.haberuygulamajava.servis.ApiClient;
import com.example.haberuygulamajava.viewmodel.KategorilerViewModel;

import retrofit2.Retrofit;

public class KategoriFragment extends Fragment {

    private FragmentKategoriBinding binding;
    private HaberlerRVAdapter adapter;
    private KategorilerViewModel viewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentKategoriBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Retrofit retrofit = ApiClient.getClient();
        HaberDao haberDao = new HaberDao(retrofit);
        viewModel = new KategorilerViewModel(haberDao);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        KategoriFragmentArgs args = KategoriFragmentArgs.fromBundle(getArguments());
        int kategoriId = args.getId();

        viewModel.loadKategoriHaberleri(kategoriId);
        observeKategoriHaber();
    }

    private void observeKategoriHaber() {
        LifecycleOwner lifecycleOwner = getViewLifecycleOwner();
        lifecycleOwner.getLifecycle().addObserver((LifecycleEventObserver) (source, event) -> {
            if (event == Lifecycle.Event.ON_START) {
                viewModel.getKategoriHaberleri().observe(lifecycleOwner, liste -> {
                    if (liste != null && !liste.isEmpty()) {
                        adapter = new HaberlerRVAdapter(requireContext(),liste, new HaberlerRVAdapter.OnItemClickListener() {
                            @Override
                            public void onItemClick(HaberModel haber) {
                                NavDirections action = KategoriFragmentDirections.kategoriToHaberDetay(haber);
                                Navigation.findNavController(binding.rvKategoriHaber).navigate(action);
                            }
                        });



                        binding.rvKategoriHaber.setLayoutManager(
                                new LinearLayoutManager(requireContext(), LinearLayoutManager.VERTICAL, false));
                        binding.rvKategoriHaber.setAdapter(adapter);
                    }
                });
            }
        });
    }
}

File: HaberUygulamaKotlin/app/src/main/java/com/example/haberuygulama/view/KategoriFragment.kt
package com.example.haberuygulama.view

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.lifecycleScope
import androidx.navigation.Navigation
import androidx.navigation.fragment.navArgs
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.haberuygulama.adapter.HaberlerRVAdapter
import com.example.haberuygulama.databinding.FragmentKategoriBinding
import com.example.haberuygulama.deo.HaberDao
import com.example.haberuygulama.servis.ApiClient
import com.example.haberuygulama.viewmodel.HaberlerViewModel
import com.example.haberuygulama.viewmodel.KategorilerViewModel


class KategoriFragment : Fragment() {

    lateinit var binding:FragmentKategoriBinding
    lateinit var adapter: HaberlerRVAdapter
    lateinit var viewModel: KategorilerViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentKategoriBinding.inflate(inflater,container,false)


        return binding.root
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = KategorilerViewModel(HaberDao(ApiClient.retrofit))

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val bundle:KategoriFragmentArgs by navArgs()
        val gelen = bundle.kategoriId
        viewModel.loadKategoriHaberleri(gelen)

        observeKategoriHaber()
    }

    private fun observeKategoriHaber() {
        lifecycleScope.launchWhenStarted {
            viewModel.kategoriHaberleri.collect { liste ->
                if (liste.isNotEmpty()) {
                    adapter = HaberlerRVAdapter(liste) {
                        // tıklanınca yapılacak işlem
                        val gecis = KategoriFragmentDirections.kategoriToHaberDetay(it)
                        Navigation.findNavController(binding.rvKategoriHaber).navigate(gecis)
                    }
                    binding.rvKategoriHaber.layoutManager =
                        LinearLayoutManager(requireContext(), LinearLayoutManager.VERTICAL, false)
                    binding.rvKategoriHaber.adapter = adapter
                }
            }
        }
    }

}
